File: examples/libgazebo/position/position.cc
#include <gazebo/gazebo.h>
#include <gazebo/GazeboError.hh>

int main()
{
  libgazebo::Client *client = new libgazebo::Client();
  libgazebo::SimulationIface *simIface = new libgazebo::SimulationIface();
  libgazebo::PositionIface *posIface = new libgazebo::PositionIface();

  int serverId = 0;

  /// Connect to the libgazebo server
  try
  {
    client->ConnectWait(serverId, GZ_CLIENT_ID_USER_FIRST);
  }
  catch (std::string e)
  {
    std::cout << "Gazebo error: Unable to connect\n" << e << "\n";
    return -1;
  }

  /// Open the Simulation Interface
  try
  {
    simIface->Open(client, "default");
  }
  catch (std::string e)
  {
    std::cout << "Gazebo error: Unable to connect to the sim interface\n" << e << "\n";
    return -1;
  }

  /// Open the Position interface
  try
  {
    posIface->Open(client, "turtlebot1::position_iface_0");
  }
  catch (std::string e)
  {
    std::cout << "Gazebo error: Unable to connect to the position interface\n"
    << e << "\n";
    return -1;
  }

  // Enable the motor
  posIface->Lock(1);
  posIface->data->cmdEnableMotors = 1;
  posIface->Unlock();

  posIface->data->cmdVelocity.yaw = -0.1;
  while (true)
  {
    posIface->Lock(1);
    posIface->data->cmdVelocity.pos.x = 0.2;
    posIface->Unlock();

    usleep(100000);
  }
  return 0;
}

File: examples/ruby/position.rb
#based on pioneer2d world
#equivalent to the libgazebo/position example

require 'gazeboc'

begin
  client = Gazeboc::Client.new 
  simIface = Gazeboc::SimulationIface.new 
  posIface = Gazeboc::PositionIface.new
rescue Exception => e  
  puts "Gazebo exceptions can be catched on Ruby"
  raise "initialization impossible, bindings installed?"
end

client.Connect 0

posIface.Open client, "pioneer2dx_model1::position_iface_0"
posIface.Lock 1
posIface.data.cmdEnableMotors = 1
posIface.Unlock


while true

  posIface.Lock 1
  posIface.data.cmdVelocity.pos.x = 0.2
  posIface.data.cmdVelocity.yaw = -0.1
  posIface.Unlock
  sleep 1
  pose = posIface.data.pose.pos
  puts "position: (%.3f, %.3f, %.3f)" % [pose.x, pose.y, pose.z]

end

posIface.Close
simIface.Close
